relational/table.py

```diff
@@ -260,3 +262,154 @@ def rows(self, proj):
+                if date_opened < oldest_open:
+                    oldest_open = date_opened
+
+            if oldest_open < date.max:
+                return (oldest_open.strftime("%Y-%m-%d"), PPTS.OUTPUT_NAME)
+
+        return ('', None)
+
+    def _entitled_date(self, rows, proj):
+        # TODO: Use the Entitlements Approved date once we have pulled
+        # in the new PPTS data pipeline (if that doesn't exist fall back)
+
+        # Look for the ENT child of a PRJ with the latest date_closed
+        # (assuming all are closed). Fall back to the PRJ date.
+        root = proj.roots[PPTS.NAME][0]
+        if root.get_latest('record_type_category')[0] == 'PRJ':
+            newest_closed = date.min
+            count_closed_no_date = 0
+            for child in proj.children[PPTS.NAME]:
+                record_type = child.get_latest('record_type_category')[0]
+                if record_type not in self._PPTS_ENT_CODES:
+                    continue
+
+                date_closed_value = child.get_latest('date_closed')
+                status_value = child.get_latest('status')
+                if date_closed_value:
+                    date_closed = datetime.strptime(
+                        date_closed_value[0].split(' ')[0],
+                        "%m/%d/%Y").date()
+                    if date_closed > newest_closed:
+                        newest_closed = date_closed
+                elif status_value and 'closed' in status_value[0].lower():
+                    count_closed_no_date += 1
+                else:
+                    # ENT record is not closed, entitlements not approved
+                    return ('', None)
+
+            if newest_closed > date.min:
+                return (newest_closed.strftime("%Y-%m-%d"), PPTS.OUTPUT_NAME)
+            elif count_closed_no_date > 0:
+                # Fall back to PRJ date if all ENT child records are closed
+                # but there's no date
+                date_closed_field = root.get_latest('date_closed')[0]
+                if date_closed_field:
+                    date_closed = datetime.strptime(
+                        date_closed_field.split(' ')[0],
+                        "%m/%d/%Y").date()
+                    return (date_closed.strftime("%Y-%m-%d"), PPTS.OUTPUT_NAME)
+
+        return ('', None)
+
+    def status_row(self,
+                   proj,
+                   top_level_status='',
+                   start_date='',
+                   end_date='',
+                   data=''):
+        row = [''] * len(self.header())
+        self.gen_id(row, proj)
+        row[self.index(self.TOP_LEVEL_STATUS)] = top_level_status
+        row[self.index(self.START_DATE)] = start_date
+        row[self.index(self.END_DATE)] = end_date
+        row[self.index(self.DATA_SOURCE)] = data
+        return row
+
+    def rows(self, proj):
+        result = []
+        (predev_date, predev_data) = self._predevelopment_date(result, proj)
```

Comment:
doesn't look like you're using result in these calls at all - remove?

Reply:
Done